src/components/AvatarGroup/AvatarGroup.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useCallback } from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import Avatar from "../Avatar/Avatar";
+import Counter from "../Counter/Counter";
+import Tooltip from "../Tooltip/Tooltip";
+import AvatarGroupCounterTooltipContainer from "./AvatarGroupCounterTooltipContainer";
+import styles from "./AvatarGroup.module.scss";
+
+const AvatarGroup = ({
+  className,
+  id,
+  children,
+  size,
+  type,
+  max,
+  counterProps,
+  counterTooltipCustomProps,
+  counterTooltipIsVirtualizedList,
+  counterTooltipTheme
+}) => {
+  const {
+    color: counterColor = Counter.colors.LIGHT,
+    count: counterValue = children?.length && children.length - max,
+    prefix: counterPrefix = "+",
+    maxDigits: counterMaxDigits = 3
+  } = counterProps || {};
+
+  const getCounterContainerSizeStyle = useCallback(() => {
+    const counterSize = size || (children?.length && children[0]?.props?.size);
+    if (counterSize && Avatar.sizes[`${counterSize.toString().toUpperCase()}`]) {
+      return styles[`${counterSize}`];
+    }
+    return styles.medium;
+  }, [children, size]);
+  const counterContainerSizeStyle = getCounterContainerSizeStyle();
+
+  const getCounterContainerColorStyle = useCallback(() => {
+    return styles[`${counterColor}`];
+  }, [counterColor]);
+  const counterContainerColorStyle = getCounterContainerColorStyle();
+
+  if (!children) {
+    return null;
+  }
+
+  if (!Array.isArray(children) && typeof children === "object") {
+    children = [children];
+  }
+
+  return (
+    <div className={cx(styles.container, className)} id={id}>
+      {children.map((avatar, index) => {
+        if (index < max) {
+          return (
+            <Avatar
+              {...avatar.props}
```

Comment:
it's not very standard. u can use:  React.cloneElement for prevent weird cases when passing a component which wrap an avatar component or something like this.